### PR TITLE
Fix - unlisted page appear in google search result.

### PR DESCRIPTION
**Pull request by AbhijitNage123**
### Description

- Fix - unlisted page appear in google search result.

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### How has this been tested?

- Unlist the page/post and check if "noindex, follow" is adding in meta or not.

### Checklist:
- [x] My code is tested
- [ ] My code passes the PHPCS tests
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: class-unlist-posts.php [hunk 1]
```diff
@@ -149,9 +150,35 @@ public function hide_post_from_searchengines() {
...
+		 * This directive tells web robots not to index the page content if the page is unlisted.
+		 *
+		 * @since  1.1.4
+		 * @param  Array $robots Associative array of robots directives.
+		 */
+		public function hide_post_page_no_robots( $robots ) {
+
+			// Bail if posts unlists is disabled.
+			if ( false === $this->allow_post_unlist() ) {
+				return false;
```
Comment (Nikschavan) [inline class-unlist-posts.php:173]: Should this be `return $robots;`?

```suggestion
				return $robots;
```

File: class-unlist-posts.php [hunk 2]
```diff
@@ -149,9 +150,35 @@ public function hide_post_from_searchengines() {
...
+		}
+
+		/**
+		 * This directive tells web robots not to index the page content if the page is unlisted.
+		 *
+		 * @since  1.1.4
+		 * @param  Array $robots Associative array of robots directives.
+		 */
+		public function hide_post_page_no_robots( $robots ) {
+
```
Comment (Nikschavan) [inline class-unlist-posts.php:170]:
```suggestion
```

File: README.md [hunk 3]
```diff
@@ -37,6 +37,9 @@ Just select option "Unlist Post" in any post of any type and that post will be h
 
 ## Changelog ##
 
+### 1.1.4 ###
+- Fix: Unlisted page appearing in google search result.
```
Comment (Nikschavan) [inline README.md:41]:
```suggestion
- Fix: due to a bug unlisted posts were visible in the search results, which we have fixed now.

```

File: class-unlist-posts.php [hunk 4]
```diff
@@ -149,9 +150,35 @@ public function hide_post_from_searchengines() {
...
+			}
+		}
+
+		/**
+		 * This directive tells web robots not to index the page content if the page is unlisted.
+		 *
+		 * @since  1.1.4
+		 * @param  Array $robots Associative array of robots directives.
+		 */
+		public function hide_post_page_no_robots( $robots ) {
```
Comment (Nikschavan) [inline class-unlist-posts.php:169]:
```suggestion
		public function no_robots_for_unlisted_posts( $robots ) {
```

File: class-unlist-posts.php [hunk 5]
```diff
@@ -62,6 +62,7 @@ public function init() {
 			add_filter( 'get_next_post_where', array( $this, 'post_navigation_clause' ), 20, 1 );
 			add_filter( 'get_previous_post_where', array( $this, 'post_navigation_clause' ), 20, 1 );
 			add_action( 'wp_head', array( $this, 'hide_post_from_searchengines' ) );
+			add_filter( 'wp_robots', array( $this, 'hide_post_page_no_robots' ) );
```
Comment (Nikschavan) [inline class-unlist-posts.php:65]:
```suggestion
			add_filter( 'wp_robots', array( $this, 'no_robots_for_unlisted_posts' ) );
```

File: class-unlist-posts.php [hunk 6]
```diff
@@ -149,9 +150,35 @@ public function hide_post_from_searchengines() {
 
 			$hidden_posts = get_option( 'unlist_posts', array() );
 
+			// wp_no_robots is deprecated since WP 5.7.
+			if ( function_exists( 'wp_robots_no_robots' ) ) {
```
Comment (patilvikasj) [inline class-unlist-posts.php:154]: Add this snippet at the top of this method.

Comment (AbhijitNage123) [reply class-unlist-posts.php]: Updated

File: class-unlist-posts.php (hunk 5 above)
Comment (AbhijitNage123) [reply class-unlist-posts.php]: Updated

File: class-unlist-posts.php (hunk 4 above)
Comment (AbhijitNage123) [reply class-unlist-posts.php]: Updated

File: README.md (hunk 3 above)
Comment (AbhijitNage123) [reply README.md]: Updated

File: class-unlist-posts.php (hunk 2 above)
Comment (AbhijitNage123) [reply class-unlist-posts.php]: Remove line space

File: class-unlist-posts.php (hunk 1 above)
Comment (AbhijitNage123) [reply class-unlist-posts.php]: Updated

**Review by patilvikasj — APPROVED** [review]

**Pull request merged**